### PR TITLE
chore: Factor out common container for editor

### PR DESCRIPTION
**Pull request by richburdon**
First stage in unification of MarkdownEditor and DocumentEditor.

- co-locates extension creation
- tighten overgeneralized large function params by removing dead code
- reduces surface area of type definitions
- improve testing/stories

Next:

- simplify type handling further in MarkdownPlugin.
- simplify extension creation.
- single container for ECHO/non-ECHO and different surface locations (section, article, etc.)

### CODE REVIEW

File: packages/plugins/plugin-markdown/src/MarkdownPlugin.tsx [hunk 1]
```diff
@@ -267,72 +243,51 @@ export const MarkdownPlugin = (): PluginDefinition<MarkdownPluginProvides> => {
...
-                  />
-                );
-              }
-              break;
-            }
+              /**
+               * Checks if an object conforms to the interface needed to render an editor.
+               */
+              // TODO(burdon): Remove need for this; intents should provide DocumentType.
+              const isEditorModel = (data: any): data is { id: string; text: string } =>
```
Comment (wittjosiah) [inline packages/plugins/plugin-markdown/src/MarkdownPlugin.tsx:254]: I think this should go back where it was, no reason it needs to be recreated in the render cycle

Comment (wittjosiah) [reply packages/plugins/plugin-markdown/src/MarkdownPlugin.tsx:254]: (same for functions below, they don't depend on any react state so no reason for them to be (re)defined inside render function)

File: packages/plugins/plugin-markdown/src/extensions.tsx [hunk 2]
```diff
@@ -2,63 +2,117 @@
...
+  viewMode?: EditorViewMode;
 };
 
-/**
- * Create extension instances for editor.
- */
-export const createBaseExtensions = ({
-  viewMode,
-  settings,
+// TODO(burdon): Merge with createBaseExtensions above.
```
Comment (wittjosiah) [inline packages/plugins/plugin-markdown/src/extensions.tsx:40]: below?

File: packages/plugins/plugin-markdown/src/extensions.tsx [hunk 3]
```diff
@@ -155,25 +206,31 @@ const onRenderLink = (onSelectObject: (id: string) => void) => (el: Element, url
...
+        classNames={style.icon}
+      />
+    </a>,
   );
 };
 
 const renderLinkTooltip = (el: Element, url: string) => {
   const web = new URL(url);
-  createRoot(el).render(
-    <StrictMode>
```
Comment (wittjosiah) [inline packages/plugins/plugin-markdown/src/extensions.tsx:172]: no more strict mode for these?

File: packages/plugins/plugin-markdown/src/extensions.tsx [hunk 4]
```diff
@@ -155,25 +206,31 @@ const onRenderLink = (onSelectObject: (id: string) => void) => (el: Element, url
...
+    <a href={url} rel='noreferrer' target='_blank' className={style.hover}>
+      {web.origin}
+      <Icon icon='ph--arrow-square-out--bold' size={4} classNames={style.icon} />
+    </a>,
   );
 };
+
+// TODO(burdon): Factor out. Reconcile with react-ui-editor.
+const renderRoot = <T extends Element>(root: T, node: ReactNode): T => {
+  createRoot(root).render(<ThemeProvider tx={defaultTx}>{node}</ThemeProvider>);
```
Comment (wittjosiah) [inline packages/plugins/plugin-markdown/src/extensions.tsx:234]: what's the difference between a new root and a portal for these things?

Comment (richburdon) [reply packages/plugins/plugin-markdown/src/extensions.tsx:234]: portals gets removed from the DOM by CM.

File: packages/plugins/plugin-markdown/src/extensions.tsx (hunk 3 above)
Comment (richburdon) [reply packages/plugins/plugin-markdown/src/extensions.tsx:172]: temporary; these will be replaced by non-react DOM directly.

**Pull request merged**